test/parallel/test-timers-same-timeout-wrong-list-deleted.js

```diff
@@ -30,41 +28,9 @@ const handle1 = setTimeout(common.mustCall(function() {
     // erroneously deleted. If we are able to cancel the timer successfully,
     // the bug is fixed.
     clearTimeout(handle2);
-
-    setImmediate(common.mustCall(function() {
-      setImmediate(common.mustCall(function() {
-        const activeTimers = getActiveTimers();
-
-        // Make sure our clearTimeout succeeded. One timer finished and
-        // the other was canceled, so none should be active.
-        assert.strictEqual(activeTimers.length, 0, 'Timers remain.');
-      }));
-    }));
   }), 1);
 
-  // Make sure our timers got added to the list.
-  const activeTimers = getActiveTimers();
-  const shortTimer = activeTimers.find(function(handle) {
-    return handle._list.msecs === 1;
-  });
-  const longTimers = activeTimers.filter(function(handle) {
-    return handle._list.msecs === TIMEOUT;
-  });
-
-  // Make sure our clearTimeout succeeded. One timer finished and
-  // the other was canceled, so none should be active.
-  assert.strictEqual(activeTimers.length, 3,
-                     'There should be 3 timers in the list.');
-  assert(shortTimer instanceof Timer, 'The shorter timer is not in the list.');
-  assert.strictEqual(longTimers.length, 2,
-                     'Both longer timers should be in the list.');
-
   // When this callback completes, `listOnTimeout` should now look at the
   // correct list and refrain from removing the new TIMEOUT list which
   // contains the reference to the newer timer.
 }), TIMEOUT);
-
-function getActiveTimers() {
-  const activeHandles = process._getActiveHandles();
-  return activeHandles.filter((handle) => handle instanceof Timer);
-}
```

Comment:
Maybe internalTimers should expose the list objects so that this test actually tests itself properly...?

I would say the case is theoretically still relevant, in concept at least.

Reply:
It should still test itself via behaviour, even without checking the handles. Unless I'm completely misunderstanding it?

Reply:
Yeah, I think you're right. The original was a bit more through but I'm also not convinced it is necessary.